### PR TITLE
[pytorch][build][sagemaker] PyTorch 1.11.0 SM training re-release, fixing perf degradation

### PR DESCRIPTION
**Pull request by ShiboXing**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
re-release of 1.11 sm image, fixing sm performance degradation

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"` commit: [3b5409d](https://github.com/aws/deep-learning-containers/pull/2220/commits/3b5409db3b55da30f2fd220a2779f52ed1722aac)
- [x] Revision B: `sagemaker_remote_tests = "rc"` commit: [8ddfafb](https://github.com/aws/deep-learning-containers/pull/2220/commits/8ddfafbd7d7e99503f7c274418d20de904ac156d)
- [x] Revision C: `sagemaker_remote_tests = "efa"` commit: [2239452](https://github.com/aws/deep-learning-containers/pull/2220/commits/2239452e78ac54cdd99f84694c8fbe1bf6f88a65)

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/buildspec.yml [hunk 1]
```diff
@@ -46,41 +46,40 @@ context:
...
-  #   tag_python_version: &TAG_PYTHON_VERSION py38
-  #   cuda_version: &CUDA_VERSION cu116
-  #   os_version: &OS_VERSION ubuntu20.04
-  #   tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION, "-", *OS_VERSION, "-ec2" ]
-  #   docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /, *CUDA_VERSION, /Dockerfile.,
-  #                        *DEVICE_TYPE ]
-  #   target: ec2
-  #   context:
-  #     <<: *TRAINING_CONTEXT
+  BuildEC2CPUPTTrainPy3DockerImage:
```
Comment (ydaiming) [inline pytorch/buildspec.yml:49]: Could you comment out the EC2 sections in this buildspec?

**Comment by ydaiming** [comment]
Could you please disable these tests for 1.10 and above?
https://github.com/aws/deep-learning-containers/blob/master/test/dlc_tests/release_candidate_integration/test_sm_profiler.py#L35

File: test/dlc_tests/release_candidate_integration/test_sm_profiler.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def test_sm_profiler_pt(pytorch_training):
         pytest.skip(f"Processor {processor} not supported. Skipping test.")
 
     _, image_framework_version = get_framework_and_version_from_tag(pytorch_training)
-    if Version(image_framework_version) in SpecifierSet(">=1.12"):
+    if Version(image_framework_version) in SpecifierSet(">=1.10"):
```
Comment (arjkesh) [inline test/dlc_tests/release_candidate_integration/test_sm_profiler.py:35]: nit: if this is expected, please change the "skip" message below

File: test/test_utils/__init__.py [hunk 3]
```diff
@@ -820,7 +820,7 @@ def upload_tests_to_s3(testname_datetime_suffix):
 
     path = run("pwd", hide=True).stdout.strip("\n")
     if "dlc_tests" not in path:
-        EnvironmentError("Test is being run from wrong path")
+        raise EnvironmentError("Test is being run from wrong path")
```
Comment (arjkesh) [inline test/test_utils/__init__.py:823]: nice catch! thank you!

**Review by josephevans — APPROVED** [review]

**Review by josephevans — DISMISSED** [review]

File: pytorch/buildspec-1-11.yml [hunk 4]
```diff
@@ -60,22 +60,22 @@ images:
...
-    tag_python_version: &TAG_PYTHON_VERSION py38
-    cuda_version: &CUDA_VERSION cu115
-    os_version: &OS_VERSION ubuntu20.04
-    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION, "-", *OS_VERSION, "-ec2" ]
-    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /, *CUDA_VERSION, /Dockerfile.,
-                         *DEVICE_TYPE ]
-    target: ec2
-    context:
-      <<: *TRAINING_CONTEXT
+  # BuildEC2GPUPTTrainPy3DockerImage:
```
Comment (tejaschumbalkar) [inline pytorch/buildspec-1-11.yml:63]: Lets revert this file change to prepare for merge

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**